docs/src/components/hardware-list.js

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import hardware from "../data/hardware.json";
+
+const compatibleLookUp = {
```

Comment:
Although `compatible` makes sense in the context of these `yml` files - as in "compatbility" or "compatible with" -  for discussions and documentation we should probably adopt more articulate terminology.

For instance, `interconnect` (as @Nicell used in his [original post](https://github.com/zmkfirmware/zmk/pull/653#issue-564559826)) or `interface` (as I used in another [thread](https://github.com/zmkfirmware/zmk/pull/653/commits/4aa3fa663716e63ba1a0dd3b9f4618afead96fa3#r574804767)) are two possible examples.  They're both similar to `pinout` but generalised to any form of shield connection.

Moreover, we should adopt terminology to differentiate between "established"/"named" schemes (e.g. the [sparkfun] `pro_micro`, `makerdiary_m2` or `adafruit_feather`) and any other interface.  Keeping in mind the points I made in https://github.com/zmkfirmware/zmk/pull/653/commits/4aa3fa663716e63ba1a0dd3b9f4618afead96fa3#r574804767.

The code should also reflect this terminology.

Thoughts?

Reply:
Anyone?  I'll elaborate a bit more if it helps ...

I'm not keen on `compatible` because it's *primarily* used as an adjective.

I personally prefer `interface` because:
a) it's a noun, which makes it easy to use in both code and sentences.
b) its dictionary definition fits the concept, making it fairly self-explanatory.
c) it has a natural/easy plural.
**but**, I suspect it *may* inadvertantly have namespace issues with some programming languages that use it as a keyword 😞

`interconnect` is a verb or adjective, so using it as a noun feels weird.

Reply:
I just noticed @petejohanson also used the word `footprint` in another comment.  Effectively a synonym of `pinout`.

Reply:
I'd be down w/ `interface`, `pinout` or `footprint`.